### PR TITLE
fix: Grid Menu is shown twice after changing frozen options

### PR DESCRIPTION
**Pull request by ghiscoding**
- when removing frozen columns and then reapplying them, we might end up with 2 Grid Menu icons, we need to make sure to delete any previous grid menu icon before recreating a new one

after calling these 2 actions multiple times, we get 2x grid menu icons

image

as shown below

image

### CODE REVIEW

**Comment by 6pac** [comment]
Hey @ghiscoding , another silly question. Whenever I pull from the master to my local repo now, I end up with:

> $ git status
> On branch master
> Your branch is up to date with 'origin/master'.
> 
> Changes not staged for commit:
>   (use "git add/rm <file>..." to update what will be committed)
>   (use "git restore <file>..." to discard changes in working directory)
>         deleted:    dist/controls/slick.columnpicker - Copy.min.js
>         deleted:    dist/controls/slick.columnpicker.js.map
>         deleted:    dist/controls/slick.gridmenu.js.map
>         deleted:    dist/controls/slick.pager.js.map
>         deleted:    dist/examples/slick.checkboxselectcolumn-test1.min.js
>         deleted:    dist/examples/slick.checkboxselectcolumn-test2.min.js
>         deleted:    dist/examples/slick.checkboxselectcolumn-test3.min.js
>         deleted:    dist/plugins/slick.autotooltips.js.map
>         deleted:    dist/plugins/slick.cellcopymanager.js.map
>         deleted:    dist/plugins/slick.cellexternalcopymanager-old.min.js
>         deleted:    dist/plugins/slick.cellexternalcopymanager.js.map
>         deleted:    dist/plugins/slick.cellmenu.js.map
>         deleted:    dist/plugins/slick.cellrangedecorator.js.map
>         deleted:    dist/plugins/slick.cellrangeselector.js.map
>         deleted:    dist/plugins/slick.cellselectionmodel.js.map
>         deleted:    dist/plugins/slick.checkboxselectcolumn.js.map
>         deleted:    dist/plugins/slick.contextmenu.js.map
>         deleted:    dist/plugins/slick.crossgridrowmovemanager.js.map
>         deleted:    dist/plugins/slick.customtooltip.js.map
>         deleted:    dist/plugins/slick.draggablegrouping.js.map
>         deleted:    dist/plugins/slick.headerbuttons.js.map
>         deleted:    dist/plugins/slick.headermenu.js.map
>         deleted:    dist/plugins/slick.resizer.js.map
>         deleted:    dist/plugins/slick.rowdetailview.js.map
>         deleted:    dist/plugins/slick.rowmovemanager.js.map
>         deleted:    dist/plugins/slick.rowselectionmodel.js.map
>         deleted:    dist/plugins/slick.state.js.map
>         deleted:    dist/slick.compositeeditor.js.map
>         deleted:    dist/slick.core.js.map
>         deleted:    dist/slick.dataview.js.map
>         deleted:    dist/slick.editors.js.map
>         deleted:    dist/slick.formatters.js.map
>         deleted:    dist/slick.grid - Copy.min.js
>         deleted:    dist/slick.grid-diag.min.js
>         deleted:    dist/slick.grid.js.map
>         deleted:    dist/slick.groupitemmetadataprovider.js.map
>         deleted:    dist/slick.interactions.js.map
>         deleted:    dist/slick.remotemodel-yahoo.js.map
>         deleted:    dist/slick.remotemodel.js.map
> 
> Untracked files:
>   (use "git add <file>..." to include in what will be committed)
>         dist/controls/slick.columnmenu.min.js.map
>         dist/controls/slick.columnpicker.min.js.map
>         dist/controls/slick.gridmenu.min.js.map
>         dist/controls/slick.pager.min.js.map
>         dist/plugins/slick.autotooltips.min.js.map
>         dist/plugins/slick.cellcopymanager.min.js.map
>         dist/plugins/slick.cellexternalcopymanager.min.js.map
>         dist/plugins/slick.cellmenu.min.js.map
>         dist/plugins/slick.cellrangedecorator.min.js.map
>         dist/plugins/slick.cellrangeselector.min.js.map
>         dist/plugins/slick.cellselectionmodel.min.js.map
>         dist/plugins/slick.checkboxselectcolumn.min.js.map
>         dist/plugins/slick.contextmenu.min.js.map
>         dist/plugins/slick.crossgridrowmovemanager.min.js.map
>         dist/plugins/slick.customtooltip.min.js.map
>         dist/plugins/slick.draggablegrouping.min.js.map
>         dist/plugins/slick.headerbuttons.min.js.map
>         dist/plugins/slick.headermenu.min.js.map
>         dist/plugins/slick.resizer.min.js.map
>         dist/plugins/slick.rowdetailview.min.js.map
>         dist/plugins/slick.rowmovemanager.min.js.map
>         dist/plugins/slick.rowselectionmodel.min.js.map
>         dist/plugins/slick.state.min.js.map
>         dist/slick.compositeeditor.min.js.map
>         dist/slick.core.min.js.map
>         dist/slick.dataview.min.js.map
>         dist/slick.editors.min.js.map
>         dist/slick.formatters.min.js.map
>         dist/slick.grid.min.js.map
>         dist/slick.groupitemmetadataprovider.min.js.map
>         dist/slick.interactions.min.js.map
>         dist/slick.remotemodel-yahoo.min.js.map
>         dist/slick.remotemodel.min.js.map
> 
> no changes added to commit (use "git add" and/or "git commit -a")

Since I never use or generate the ```dist``` folder (I use VS with built in minifying and bundling, and never use NPM), I'm not quite sure what I should be doing. In the past I may have deleted the files but since I'm about to release a version I'm checking that everything is set up correctly before I do it. Do you have any tips about what I should be doing to correctly resolve the above?

**Comment by ghiscoding** [comment]
in my own repos I never keep the `dist` folder, however my repos only work with npm (you need to npm install my repos to work with them), however with SlickGrid we need to support the legacy way which are standalone `<script>` and for that reason I added the `dist` into the git project. This legacy approach is also what we use for some of the examples (since we're not building any single page app and we're using standalone script everywhere), which by the way the next major will also have this approach of keeping the `dist` folder and will be the only approach actually so we're kinda stuck with that... What I see in your list though is that there's a lot of invalid files, a bunch of "copy" and a few "test" files that should be deleted

Anyway in short, you should just be able to run the npm script to minify all JS files (.min) and I think the minify is also called by the release script (yes it does I just checked). The release will also add all min files to git automatically so it's transparent to you but if you run the minify script then you would have to add them to git yourself

```sh
npm run minify    # will minify all JS files
nom run release   # will minify all files and then publish new release
```
every time you run the minify, it will update most of dist files (unless there's nothing that changed, if so the minified file shouldn't detect any change in git)

does that answer your question?

**Comment by 6pac** [comment]
OK. I tried minifying and the list just got longer. 

I had a look and this folder is tracked in the repo, so I think the easiest is for me just to make sure everything local is synced with the repo and then download the repo as a ZIP file from GitHub and simply replace the whole ```dist``` folder with what's in there. That will guarantee no local changes.

I don't really care what's in the ```dist``` folder, and it looks like it will be kept up to date by the automations, so I mainly don't want to add or remove any files from the tracking index that might break something or add rubbish.

**Comment by 6pac** [comment]
Note I'm only talking about tidying things up pre doing a release, I wouldn't worry about it otherwise.

**Comment by ghiscoding** [comment]
I'm not sure what you mean exactly but the minify process simply minify all the JS and CSS files. It shouldn't add any extra files unless you add JS or CSS files anywhere else in the project (for example all these "copy x" or "text x" JS files that I see coming up in the dist folder, these Copy files might be skipped by the `.gitignore` but since the `minify` process minifies everything it finds then it does add those in the dist folder).

There's maybe 1 thing I forgot to do which is to run the "clean" npm script before calling the "minify" npm script. 

https://github.com/6pac/SlickGrid/blob/b36720937d7276ed344979edb83b0e281a75d9df/package.json#L27-L32

basically you could try to add this line as a pre script which will clean the dist folder before you run the minify process

```diff
"scripts": {
  //...
+ "preminify": "npm run clean",
    "minify": "node ./scripts/minify.js --start",
```

**Comment by 6pac** [comment]
OK will try. Yeah, I'm aware of what the minify script does, and the purpose of the ```dist``` folder, but I'm just not sure why my local files are all being flagged as a conflict with GitHub, even when I sync up and freshly minify them. 

Even copying the files from the repo didn't work!?

Finally I used:
 
    git stash push --include-untracked
    git stash drop
    git pull --no-rebase

and that worked. Will try the ```npm run clean``` next time around.
